torch/csrc/jit/docs/OVERVIEW.md

```diff
@@ -548,7 +549,97 @@ This makes it possible to use most of the compiler functionality when python is
 
 [script/compiler.cpp](../script/compiler.cpp)
 
-The Environment object tracks the assignment of variable names to SugaredValues during compilation. It is local to the compiler file. A stack of environments exist, with a new environment being created for sub-blocks introduced by control flow. The Environment also handles turning the AST representation into SSA-form by tracking which variables were modified inside a sub-block and inserting the correct inputs/outputs to the Blocks of if-statements and loops.
+The Environment object tracks the assignment of variable names during compilation. It is local to the compiler file. A stack of environments exist, with a new environment being created for sub-blocks introduced by control flow. The Environment keeps two tables, one for values which are not first class in the type system (Sugared values) and a type table for values which are. When first class values are set, we emit a prim::Store, and when they are referenced we emit a prim::Load. Sugared values are not re-assignable. The graph is converted to SSA in the convertToSSA pass.
+
+## Conversion To SSA ##
+
+[script/convert_to_ssa.cpp](../script/convert_to_ssa.cpp)
+
+Converting to SSA works in multiple parts.
+- First, we add loads and stores to control flow operators (ifs & loops).
+- Then we erase Break & Continue statements from the graph and replace them with `prim::LoopContinuation`. `prim::LoopContinuation` has the form `LoopContinuation(%loop_continue_condition, %loop_carried_vars)`. Break Statements have the continue condition set to false, and Continue statements inline the loop condition. %loop_carried_vars are the loop carried variables of the inner most loop that contains the Break or Continue statement, are added by inserting prim::Loads calls at the location of the statement.
+- Then we inline the loop condition into the graph loops.
+- Next we erase loads and stores, removing all Stores and replacing all loads
+with whatever the in-scope value of the variable name is.
```

Comment:
this seems to be a little bit too dry. How are we tracking in-scope values? Are we doing something similar to what we used to do in `compiler.cpp` before? Ideally, we could use a simple example that shows a graph with loads and stores and then the one w/o it right before we remove `prim::LoopContinuations`.